src/compaction/drop_range.rs

```diff
@@ -34,7 +64,10 @@ impl CompactionStrategy for Strategy {
             .current_version()
             .iter_levels()
             .flat_map(|lvl| lvl.iter())
-            .flat_map(|run| run.get_contained(&self.key_range))
+            .flat_map(|run| {
+                run.iter()
+                    .filter(|segment| self.bounds.contains(segment.key_range()))
+            })
```

Comment:
It's a bit unfortunate to lose `Run::get_contained` here because that actually uses binary search instead of a linear filter, which is nicer when dropping small ranges from large levels (think 1000+ tables).
So that might be worth bringing back, as the compaction strategies run while holding a write lock, so they need to be somewhat quick.

Reply:
updated the code to use the binary search code path for initial filtering